packages/manager/modules/vps/src/dashboard/migration-schedule/vps-dashboard-migration-schedule.routing.js

```diff
@@ -0,0 +1,43 @@
+import scheduleComponent from '../../migration/components/plan/plan.component';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('vps.detail.dashboard.schedule', {
+    url: '/migration/schedule',
+    component: scheduleComponent.name,
+    translations: {
+      value: ['.'],
+      format: 'json',
+    },
+    layout: 'modal',
+    redirectTo: (transition) =>
+      transition
+        .injector()
+        .getAsync('servers')
+        .then((servers) =>
+          servers && servers.length > 0
+            ? false
+            : {
+                state: 'vps.detail.dashboard',
+              },
+        ),
```

Comment:
Hmm is it a way to secure access to this modal ? I mean, the user will be prompted by a something which says "your VPS should be migrated", he then tries to access the modal, and is redirected if there is no server ?

Reply:
This modal is opened from the vps dashboard. This modal opens on a new route and the servers parameter is passed programatically.
If the user visits the new route of the modal directly, the behavior would inconsistent. Hence the above check is to make sure that the user is not able to directly visit the route.